### PR TITLE
Change the name of the flag for fast estimator in `nk.experimental.VarianceObservable`

### PR DESCRIPTION
**Pull request by alleSini99**
Modifying the flag for using the faster and biased estimator for computing the quantum variance of an arbitrary operator.

### CODE REVIEW

**Comment by PhilipVinc** [comment]
The name of the flag should be discussed a bit and a few options should be proposed with pros/cons before a candidate shall be accepted.

Also, regardless of what we stick with the old one should throw a deprecation warning.

**Comment by alleSini99** [comment]
Among all the proposals for the flag:
- `use_fast_estimator`
- `use_Oloc2`
- `use_biased_estimator`
- `use_Oloc_squared`
- `use_local_energy_squared`

the ones that obtain more votes are `use_Oloc2` and `use_Oloc_squared`. Since the first one was the original one and apparently generated some misunderstandings, I propose to use `use_Oloc_squared`.

File: netket/experimental/observable/variance/variance_operator.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def __init__(self, operator: AbstractOperator, use_fast_estimator: bool = False)
 
         Args:
             operator: The operator for which the variance is to be computed.
-            use_fast_estimator: If False, uses the local estimator of the squared operator `O^2` for variance computation (defaults to False).
+            use_Oloc_squared: If False, uses the local estimator of the squared operator `O^2` for variance computation (defaults to False).
```
Comment (PhilipVinc) [inline netket/experimental/observable/variance/variance_operator.py:52]: Please change to something like  '(Defaults False) if True, uses the fast estimator obtained by squaring the local estimator O^loc. If False, uses the straightforward estimator of (O@O)_loc which is quadratically more expensive. The fast estimator can sometimes lead to worse quality results.

Comment (alleSini99) [reply netket/experimental/observable/variance/variance_operator.py]: Biensûr Filippo

**Review by PhilipVinc — APPROVED** [review]

**Pull request merged**